storage/innobase/log/log0recv.cc

```diff
@@ -1714,11 +1714,10 @@ recv_recover_page_func(
 
 	ut_a(recv_sys->n_addrs > 0);
 	if (ulint n = --recv_sys->n_addrs) {
-		if (recv_sys->report(time)) {
+		if (report(&recv_sys->progress_time, time)) {
 			ib_logf(IB_LOG_LEVEL_INFO,
 				"To recover: " ULINTPF " pages from log", n);
-			sd_notifyf(0, "STATUS=To recover: " ULINTPF
-				   " pages from log", n);
+		        my_sd_extend_timeout(INNODB_EXTEND_TIMEOUT_INTERVAL);
```

Comment:
I think that the progress reporting was useful. Can we issue both messages? Will systemd display the startup progress if the timeout is being extended?

Please make `recv_sys_t::report()` invoke the new `report()` function.

Reply:
certainly can do both. I'm creating a IB_LOG_LEVEL_INFO_SYSTEMD_STATUS level that does both to save duplication.

I've removed `recv_sys_t::report` in favour of the new `report()`. Didn't seem much point keeping it around for 2 uses.